feat: add cfn-lint

vishnu-deepsource opened the pull request.

vishnu-deepsource committed:
feat: add cfn-lint

deepsource-autofix[bot] committed:
style: format code with Black and isort

This commit fixes the style issues introduced in ff2612f according to the output
from Black and isort.

Details: https://github.com/DeepSourceCorp/community-analyzers/pull/8

vishnu-deepsource committed:
fix: update issue categories

srijan-deepsource commented on analyzers/cfn-lint/.deepsource/analyzer/example_config.toml:
```diff
@@ -0,0 +1,4 @@
+version = 1
+
+[[analyzers]]
+name = "cfn-lint"
```
```suggestion
name = "cfn-lint"
type = "community"
```

srijan-deepsource commented on analyzers/cfn-lint/.deepsource/analyzer/issues/CFLIN-E0002.toml:
issues folder should be outside. `.deepsource/issues`

srijan-deepsource commented on analyzers/cfn-lint/utils/issue_gen.py:
```diff
@@ -0,0 +1,165 @@
+
+
+if __name__ == "__main__":
+    base_directory = "/Users/vishnu/Code/deepsource/"
+    rules_directory = "./cfn-lint/src/cfnlint/rules"
```
Remove hardcoded values? We can take this in as input.

srijan-deepsource commented on analyzers/cfn-lint/.deepsource/analyzer/issues/CFLIN-W8001.toml:
```diff
@@ -0,0 +1,11 @@
+title = "Check if Conditions are Used"
```
```suggestion
title = "Unused conditions detected"
```

srijan-deepsource commented on analyzers/cfn-lint/.deepsource/analyzer/issues/CFLIN-W8003.toml:
```diff
@@ -0,0 +1,11 @@
+title = "Fn::Equals will always return true or false"
```
```suggestion
title = "`Fn::Equals` will always return true or false"
```

srijan-deepsource reviewed (changes requested):
I see that a good number of titles are in title-case. We use sentence-case for titles. Can you please take care of them?

Also, add a test for cfn lint's sarif file to make sure our sarif-parser works properly with this.
Ref: https://github.com/DeepSourceCorp/community-analyzers/tree/master/tests

vishnu-deepsource committed:
chore: move issues to correct directory

vishnu-deepsource committed:
Update analyzers/cfn-lint/.deepsource/analyzer/example_config.toml

Co-authored-by: Srijan Saurav <68371686+srijan-deepsource@users.noreply.github.com>
Signed-off-by: Vishnu Jayadevan <68588475+vishnu-deepsource@users.noreply.github.com>

vishnu-deepsource committed:
fix: improve issue generation script

deepsource-autofix[bot] committed:
style: format code with Black and isort

This commit fixes the style issues introduced in 7034f03 according to the output
from Black and isort.

Details: https://github.com/DeepSourceCorp/community-analyzers/pull/8

vishnu-deepsource committed:
feat: Add Github Action

vishnu-deepsource committed:
feat: add tests

srijan-deepsource committed:
chore: add an end to end parsing test (#10)

Changes:
  - Add end to end parsing test
  - Fix assets being copied in the Dockerfile
  - Explicitly log the info passed to sentry.raise_info
  - Add instruction about adding fixture for parsing test in readme
  - Change workdir
  - fix dependencies installation in the image

srijan-deepsource committed:
fix: don't duplicate issues when same artifact is reported again (#12)

Changes:
- Check the artifact result's hash before parsing. Ignore the artifact
if the hash is same.
  - Raise sentry info alerts for unsanitised issues
  - fix type hints
  - add missing tests

srijan-deepsource committed:
chore: add armory (#13)

Armory is an in-house github action to validate the Analyzer directories
and issues.
It highlights the issues in the PRs that can lead to unexpected
behaviours during analyzer syncs.

---------

Signed-off-by: Srijan Saurav <68371686+srijan-deepsource@users.noreply.github.com>

srijan-deepsource committed:
chore: fail tests if the test sarif reports have unsanitised issue codes (#14)

srijan-deepsource committed:
fix: always raise issues in files sent for analysis (#15)

eshaan-deepsource committed:
feat: add slither (#9)

Slither: https://github.com/crytic/slither

---------

Co-authored-by: Srijan Saurav <68371686+srijan-deepsource@users.noreply.github.com>

vishnu-deepsource committed:
feat: add tests

vishnu-deepsource committed:
feat: add issue_map.json and generator

deepsource-autofix[bot] committed:
style: format code with Black and isort

This commit fixes the style issues introduced in 659eec1 according to the output
from Black and isort.

Details: https://github.com/DeepSourceCorp/community-analyzers/pull/8

vishnu-deepsource committed:
fix: minor mistake in toml generation

vishnu-deepsource committed:
fix: deepsource issues

vishnu-deepsource committed:
Merge branch 'master' into cfn-lint

vishnu-deepsource committed:
fix: minor DeepSource issue

vishnu-deepsource committed:
fix: deepsource issues and docstrings

vishnu-deepsource committed:
fix: deepsource issues and docstrings

eshaan-deepsource commented on analyzers/cfn-lint/CI/github-actions.yaml:
```diff
@@ -0,0 +1,42 @@
+          curl https://deepsource.io/cli | sh
+
+          # Send the report to DeepSource
+          ./bin/deepsource report --analyzer cfn-lint --value-file ./cfn-lint.sarif
```
```suggestion
          ./bin/deepsource report --analyzer cfn-lint --analyzer-type community --value-file ./cfn-lint.sarif
```

eshaan-deepsource commented on analyzers/cfn-lint/.deepsource/issues/CFLIN-E0002.toml:
```diff
@@ -0,0 +1,12 @@
+title = "error processing rule on the template"
```
Let's make all the titles capitilized.

vishnu-deepsource committed:
Update analyzers/cfn-lint/CI/github-actions.yaml

Co-authored-by: Eshaan Bansal <eshaan@deepsource.io>
Signed-off-by: Vishnu Jayadevan <68588475+vishnu-deepsource@users.noreply.github.com>

vishnu-deepsource committed:
fix: fix capitalization of issue titles

eshaan-deepsource commented:
Can you add an entry for cfn-lint in the [README.md](https://github.com/DeepSourceCorp/community-analyzers?tab=readme-ov-file#supported-analyzers) as well?

vishnu-deepsource committed:
chore: update README.md
